main.js

```diff
@@ -1,7 +1,120 @@
 /* eslint-disable no-unused-vars */
 import React from 'react'
 import ReactDOM from 'react-dom'
-import FilteredPage from './index'
+import APInput from './components/APInput.js'
+import Article from './components/Article.js'
+import Saved from './components/Saved.js'
+import Select from './components/Select.js'
+
+const SECTIONS = ['home', 'opinion', 'world', 'national', 'politics', 'upshot', 'nyregion', 'business', 'technology', 'science', 'health', 'sports', 'arts', 'books', 'movies', 'theater', 'sundayreview', 'fashion', 'tmagazine', 'food', 'travel', 'magazine', 'food', 'travel', 'magazine', 'realestatem', 'automobiles', 'obituaries', 'insider', 'saved']
+const apiURL = 'https://api.nytimes.com/svc/topstories/v2/'
+const apiSECTION = 'home.json'
+const apiKEY = '?api-key='
+const buildAPI = (apiURL + apiSECTION + apiKEY)
+const myStorage = window.localStorage
+
+class FilteredPage extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      apiInput: '',
+      result: [],
+      saved: []
+    }
+    this.handleSelectChange = this.handleSelectChange.bind(this)
+    this.handleInputChange = this.handleInputChange.bind(this)
+    this.handleInputSubmit = this.handleInputSubmit.bind(this)
+    this.handleQueue = this.handleQueue.bind(this)
+    this.handleRemove = this.handleRemove.bind(this)
+  }
+  handleInputChange(e) {
+    this.setState({
+      apiInput: e.target.value
+    })
+  }
+  handleInputSubmit() {
+    fetch(buildAPI + this.state.apiInput)
+      .then(res => res.json())
+      .then(res =>
+        this.setState({
+          result: res.results
+        }))
+      .catch(err =>
+        console.log(err)
+      )
+  }
+  handleSelectChange(e) {
+    if (e.target.value === 'saved') {
+      this.setState({
+        result: []
+      })
+    }
+    else {
+      fetch(apiURL + e.target.value + '.json' + apiKEY + this.state.apiInput)
+        .then(res => res.json())
+        .then(res =>
+          this.setState({
+            result: res.results
+          }))
+    }
+  }
+  handleQueue(result) {
+    myStorage.setItem(result.title, JSON.stringify(result))
+    this.setState({
+      saved: Object.keys(myStorage).map(result =>
```

Comment:
Notice the repetition of this statement:

```js
this.setState({
  saved: Object.keys(myStorage).map(result =>
    JSON.parse(myStorage.getItem(result))
  )
})
```

You could extract it into a method like:
```js
loadSaved() {
  this.setState({
    saved: Object.keys(myStorage).map(result =>
      JSON.parse(myStorage.getItem(result))
    )
  })
}
```

And just call your method.